ibis/backends/tests/test_impure.py

```diff
@@ -0,0 +1,67 @@
+from __future__ import annotations
+
+import pandas.testing as tm
+import pytest
+
+import ibis
+from ibis import _
+
+_counter = 0
+
+
+@ibis.udf.scalar.python(side_effects=True)
+def get_id() -> int:
+    global _counter  # noqa: PLW0603
```

Comment:
I think it's sufficient to invoke `random.random()` here. The `global` counter is unnecessarily complicated.

Reply:
The other consideration is that unless the database provides a flag for informing it that the UDF is impure, there's nothing Ibis can do to prevent incorrect assumptions based on use of global state.

You should really avoid this kind of assumption/architecture in general.

Still, we can make sure that UDFs are effectively assumed to behave as impure functions by default.

Reply:
hmm right that is way better